src/Symfony/Component/HttpClient/Internal/HttplugClientTrait.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Internal;
+
+use Http\Client\Exception\NetworkException;
+use Http\Client\Exception\RequestException;
+use Http\Client\HttpClient;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Client\NetworkExceptionInterface;
+use Psr\Http\Client\RequestExceptionInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+if (is_subclass_of(HttpClient::class, ClientInterface::class)) {
+    trait HttplugClientTrait
+    {
+        public function sendRequest(RequestInterface $request): ResponseInterface
```

Comment:
isn't it allowed to always add the return type on the method ? Then we don't need this trait

Reply:
Not on php 7.1 unfortunately.
On master yes!

Reply:
Wait, you're right, PHP 7.0 allows that too!